File: app/src/main/java/promax/dohaumen/text_edittor_mvvm/data/file_text/FileTextDao.java
package promax.dohaumen.text_edittor_mvvm.data.file_text;


import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;
import promax.dohaumen.text_edittor_mvvm.models.FileText;

/**
 * 19/11/2020
 * Room trên kotlin không dùng được, cuối cùng phải chuyển lên Java lại ngon
 */
@Dao
public interface FileTextDao {
    @Insert
    void insert(FileText... fileText);

    @Update
    void update(FileText... fileText);

    @Delete
    void delete(FileText... fileText);

    @Query("DELETE FROM filetext")
    void deleteAll();

    @Query("SELECT * FROM filetext WHERE isDeleted = :getListDelete")
    List<FileText> getList(boolean getListDelete);

    @Query("SELECT * FROM filetext WHERE isDeleted = :getListDelete")
    LiveData<List<FileText>> getLiveData(boolean getListDelete);

}

File: app/src/main/java/promax/dohaumen/text_edittor_mvvm/data/FileTextDao.kt
package promax.dohaumen.text_edittor_mvvm.data

import androidx.lifecycle.LiveData
import androidx.room.*
import promax.dohaumen.text_edittor_mvvm.models.FileText

@Dao
interface FileTextDao {
    @Query("SELECT * FROM filetext")
    fun getListText(): MutableList<FileText>

    @Query("SELECT * FROM filetext")
    fun getLiveData(): LiveData<MutableList<FileText>>

    @Update
    fun update(fileText: FileText)

    @Delete
    fun detele(fileText: FileText)

    @Insert
    fun insert(vararg fileText: FileText)

    @Query("DELETE FROM filetext")
    fun deleteAll()

}